pkg/controller/deployment/deployment_controller.go

```diff
@@ -64,24 +63,71 @@ func (d *DeploymentController) reconcileDeployments() error {
 }
 
 func (d *DeploymentController) reconcileDeployment(deployment *experimental.Deployment) error {
-	targetedRCs, err := d.getTargetedRCs(deployment)
+	switch deployment.Spec.Strategy.Type {
+	case experimental.RecreateDeploymentStrategyType:
+		return d.reconcileRecreateDeployment(*deployment)
+	case experimental.RollingUpdateDeploymentStrategyType:
+		return d.reconcileRollingUpdateDeployment(*deployment)
+	}
+	return fmt.Errorf("Unexpected deployment strategy type: %s", deployment.Spec.Strategy.Type)
+}
+
+func (d *DeploymentController) reconcileRecreateDeployment(deployment experimental.Deployment) error {
+	// TODO: implement me.
+	return nil
+}
+
+func (d *DeploymentController) reconcileRollingUpdateDeployment(deployment experimental.Deployment) error {
+	desiredRC, err := d.getDesiredRC(deployment)
```

Comment:
I'm not a huge fan of desired and targeted as the names for the RC's - while it works, I feel like you could easily make a case that they mean the opposite things (desired could mean desired to reduce, targeted could be targeted to increase). How about just oldRCs and newRC? It's not 100% perfect but it feels much more intuitive to keeping track of what's going on.